File: python/problems/triangle.py
import math

class Triangle:
    def __init__(self, aIn, bIn, cIn):

        if (aIn <= 0 or bIn <= 0 or cIn <= 0):
            raise Exception("Cannot form this triangle because the sides are negative.")
        if ((aIn >= bIn + cIn) or (bIn >= aIn + cIn) or (cIn >= aIn + bIn)):
            raise Exception("Cannot form this triangle because the sides would not connect")

        self.sideA = aIn
        self.sideB = bIn
        self.sideC = cIn

    def classify(self):
        if self.sideA == self.sideB == self.sideC:
            return "equilateral"
        elif self.sideA == self.sideB or self.sideA == self.sideC or self.sideB == self.sideC:
            return "isosceles"
        else:
            return "scalene"


if __name__ == '__main__':
    eq_test = Triangle(3,3, 3)
    is_test = Triangle(3,3,math.sqrt(18))
    sc_test = Triangle(3,4,5)

    print(eq_test.classify())
    print(is_test.classify())
    print(sc_test.classify())

File: java/java_2/labs/lab1/Triangle.java
package cmsc256;

/** 
  Determines if three doubles can be sides of triangle. 
*/
public class Triangle  {

   private double sideA;            /** Length of side 1. */
   private double sideB;            /** Length of side 2. */
   private double sideC;            /** Length of side 3. */

   /**
    * Creates a Triangle.
    *
    * @param aIn length of side 1.
    * @param bIn length of side 2.
    * @param cIn length of side 3.
    */
   public Triangle(double aIn, double bIn, double cIn) {
      
      if (aIn <= 0 || bIn <= 0 || cIn <= 0) {
         throw new IllegalArgumentException("Sides: " + aIn + " " + bIn + " " + cIn
            + " cannot form a triangle because sides cannot be negative");       // add an explanation why this exception is being thrown.
      }
     
      if ((aIn >= bIn + cIn) || (bIn >= aIn + cIn) || (cIn >= aIn + bIn)) {
         throw new IllegalArgumentException("Sides: "
            + aIn + " " + bIn + " " + cIn
            + " cannot form a triangle because the sides would not connect");      // add an explanation why this exception is being thrown.
      }
     
      sideA = aIn;
      sideB = bIn;
      sideC = cIn;
   }

   /**
    *  Classifies a triangle based on the lengths of the three sides.
    *  The classifications are either: "equilateral", "scalene", "isosceles", 
    *  or "not a triangle". The returned string must match one of these.Tri
    * 
    * @return the triangle classification.
    */
   public String classify() {
      String result = "";
      //TODO:  implement this method according to the specification in the comment block

      if (sideA == sideB && sideA == sideC){
         result += "equilateral";
      }
      else if(sideA == sideB || sideA == sideC || sideB == sideC){
         result += "isosceles";
      } else{
         result += "scalene";
      }
      return result;
   }

}
